cpp/src/io/avro/reader_impl.cu

```diff
@@ -309,8 +310,11 @@ rmm::device_buffer decompress_data(datasource& source,
     CUDF_EXPECTS(status == nvcompStatus_t::nvcompSuccess,
                  "Unable to get scratch size for snappy decompression");
 
-    rmm::device_buffer scratch(temp_size, stream);
-    rmm::device_buffer decomp_block_data(uncompressed_data_size, stream);
+    // Buffer needs to be padded.
+    rmm::device_buffer scratch(cudf::util::round_up_safe(temp_size, BUFFER_PADDING_MULTIPLE),
+                               stream);
+    rmm::device_buffer decomp_block_data(
+      cudf::util::round_up_safe(uncompressed_data_size, BUFFER_PADDING_MULTIPLE), stream);
```

Comment:
this one is odd, there should be no padding requirement from nvcomp. Have you verified that padding is required in the snappy case?

Reply:
Why not? It is input to `nvcompBatchedSnappyDecompressAsync`. 
Indeed, we don't have any unit test for AVRO in libcudf thus I can't verify it.

Reply:
Just checked, there are no alignment/padding requirements from nvcomp here. IMO this part of the change can be reverted.

Are cuDF tests useless in this context?

Reply:
Then I'll revert this.

> Are cuDF tests useless in this context?
There is 0 (zero) C++ test for avro.

Reply:
I was referring to Python tests, we have some limited coverage there.

Reply:
We have `avro_reader_test` in Python but I don't know how to check it with `compute-sanitizer`.